action.yml

```diff
@@ -55,9 +59,15 @@ runs:
           VERSION=
         fi
 
+        # If the dagger commit is 'latest', set the version back to 'head'.
+        COMMIT=${{ inputs.commit }}
+        if [[ "$VERSION" == "latest" ]]; then
+          VERSION="head"
+        fi
```

Comment:
The best approach would be to pass both values as they are and trust `install.sh` to do the right thing.

`head` only makes sense in the context of `COMMIT` as in _the `HEAD` commit for the `main` branch_, i.e.

```console
❯ curl -I https://dl.dagger.io/dagger/main/head/checksums.txt
HTTP/2 200
content-type: text/plain; charset=utf-8
content-length: 1838
date: Tue, 22 Oct 2024 15:09:24 GMT
last-modified: Tue, 22 Oct 2024 12:46:28 GMT
etag: "8b336b45f49a98e101aaa43ca402ae32"
x-amz-server-side-encryption: AES256
content-disposition: attachment;filename=checksums.txt
accept-ranges: bytes
server: AmazonS3
x-cache: Miss from cloudfront
via: 1.1 01c1372965efe3974af81a7941e07b0c.cloudfront.net (CloudFront)
x-amz-cf-pop: LHR5-P7
alt-svc: h3=":443"; ma=86400
x-amz-cf-id: kYBZp0pDrxYjecpPfJFjnAgNmxUTFZUQhHPXBNQ3qZujDOmjLdPWKw==
```

`latest` only makes sense in the context of `VERSION` as in _the latest version_, i.e.

```console
❯ curl -I https://dl.dagger.io/dagger/versions/latest
HTTP/2 200
content-type: binary/octet-stream
content-length: 7
date: Tue, 22 Oct 2024 15:10:49 GMT
last-modified: Thu, 10 Oct 2024 22:27:36 GMT
etag: "818b6f96a7646a5e9e0ede3d9a27e78d"
x-amz-server-side-encryption: AES256
accept-ranges: bytes
server: AmazonS3
x-cache: Miss from cloudfront
via: 1.1 f5db953762dd20ad78d8b2c1e8d66550.cloudfront.net (CloudFront)
x-amz-cf-pop: LHR5-P7
alt-svc: h3=":443"; ma=86400
x-amz-cf-id: Dm3qHYxI8k6fqtsnIrBvV4ZuLGtrktmJWD6Bx_0y3DEveUt6ukbZpg==
```

Expected behaviour:
1. `version` defaults to a specific version
2. `version` can be set explicitly, e.g. `0.13.4`
3. `commit` is empty by default
4. if `commit` is set, it takes precedence over `version`
5. `commit` can be set to a specific git sha
6. `commit` can be set to `head`

Reply:
> `head` only makes sense in the context of `COMMIT` as in _the `HEAD` commit for the `main` branch_, i.e.

Done.

> The best approach would be to pass both values as they are and trust `install.sh` to do the right thing.

We can't do this yet - we need to update `install.sh` to handle "latest" if we want to do this.

```bash
$ curl -fsSL https://dl.dagger.io/dagger/install.sh | DAGGER_VERSION=latest sh
sh debug downloading files into /tmp/tmp.gWarmHq8ep
sh debug http_download https://dl.dagger.io/dagger/releases/latest/dagger_vlatest_linux_amd64.tar.gz
sh debug http_download_curl received HTTP status 403
```